src/policies/security.ts

```diff
@@ -0,0 +1,233 @@
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-proc-mount.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    const hasOtherProcMount = containers(request)
+      .flatMap(c => c.securityContext || {})
+      .find(ctx => ctx.procMount !== "Default");
+
+    if (hasOtherProcMount) {
+      return request.Deny(`Changing the proc mount from the default is not allowed.`);
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * Restrict Seccomp in Pods
+ *
+ * The SecComp profile should not be explicitly set to Unconfined. This policy ensures
+ * that the `seccompProfile.Type` is undefined or restricted to `RuntimeDefault` or
+ * `Localhost`. Applies to Pods and all types of containers within them.
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-seccomp.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    const authorizedTyes = ["RuntimeDefault", "Localhost"];
+
+    // Check Pod level security context
+    const podProfileType = request.Raw.spec?.securityContext?.seccompProfile?.type || "";
+    if (!authorizedTyes.includes(podProfileType)) {
+      return request.Deny(`The seccomp profile at Pod level is not in the allowed list.`);
+    }
+
+    const hasOtherSecProfileType = containers(request)
+      .flatMap(c => c.securityContext?.seccompProfile)
+      .find(profile => !authorizedTyes.includes(profile?.type || ""));
+
+    if (hasOtherSecProfileType) {
+      return request.Deny(`Unauthorized seccomp profile type.`);
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * Restrict SELinux Type in Pods
+ *
+ * SELinux options can be used to escalate privileges. This policy ensures that the
+ * `seLinuxOptions` type field is set to either `container_t`, `container_init_t`, or
+ * `container_kvm_t`. Applies to Pods and all types of containers within them.
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-selinux-type.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    if (exemptSELinuxTypes(request)) {
+      return request.Approve();
+    }
+
+    const allowedSeLinuxTypes = ["container_t", "container_init_t", "container_kvm_t"];
+
+    // Check Pod level security context
+    const podSeLinuxType = request.Raw.spec?.securityContext?.seLinuxOptions?.type || "";
+    if (!allowedSeLinuxTypes.includes(podSeLinuxType)) {
+      return request.Deny(`Setting SELinux type at Pod level is restricted to the allowed list.`);
+    }
+
+    const hasOtherSELinuxType = containers(request)
+      .flatMap(c => c.securityContext?.seLinuxOptions?.type)
+      .find(t => !allowedSeLinuxTypes.includes(t || ""));
+
+    if (hasOtherSELinuxType) {
+      return request.Deny(`Unauthorized SELinux type.`);
```

Comment:
nit: same comment on listing that containers are the problem to be more verbose.

Reply:
Might also be valuable to list the specific container(s) that violate, that was always a painpoint I had with kyverno, etc (just seeing a generic "your pod is bad". Not sure if that is too much to add in the logic though.